jaeger-production-template.yml

```diff
@@ -163,3 +169,68 @@ items:
     selector:
       jaeger-infra: query-pod
     type: LoadBalancer
+- apiVersion: apps/v1
+  kind: DaemonSet
+  metadata:
+    name: jaeger-agent
+    labels:
+      app: jaeger
+      jaeger-infra: agent-daemonset
+  spec:
+    selector:
+      matchLabels:
+        app: jaeger
+        jaeger-infra: agent-instance
+    template:
+      metadata:
+        labels:
+          app: jaeger
+          jaeger-infra: agent-instance
+      spec:
```

Comment:
Needs a `hostnetwork: true` in here somewhere

Reply:
Why's that? Do we want the agent to receive spans from outside of the Kubernetes cluster?

Reply:
Also need to make sure the agent listens on the interface IP, not localhost (again, because you can't route to `localhost` from the pod)

Reply:
I'm missing something here. I'm under the assumption that the target applications are *not* going to send spans to the agent using `localhost` as its address (as opposed to how it happens with a sidecar or bare metal deployment). Rather, they would send spans to a known address, like this:

https://github.com/jpkrohling/origin/blob/JPK-AddedJaegerTracingToHelloWorld/examples/hello-openshift/hello_openshift.go#L43